addons/im_livechat/static/src/core/web/discuss_client_action_patch.js

```diff
@@ -0,0 +1,26 @@
+/* @odoo-module */
+
+import { DiscussClientAction } from "@mail/core/web/discuss_client_action";
+import { patch } from "@web/core/utils/patch";
+import { useService } from "@web/core/utils/hooks";
+
+patch(DiscussClientAction.prototype, {
+    setup() {
+        super.setup();
+        this.menu = useService("menu");
+        const menuId = this.props.action.params?.menu_id;
+        const appId = (this.menu.getCurrentApp() || this.menu.getMenu(menuId))?.xmlid;
+        this.store.discuss.fromLivechatMenu = appId === "im_livechat.menu_livechat_root";
```

Comment:
I'm not fan of this `fromLivechatMenu` variable (and the code that depends on it), and it's unreliable as well.